File: 0125-valid-palindrome/0125-valid-palindrome.cpp
class Solution {
public:
    bool isPalindrome(string s) {
        if(s.length()==1) return true;
        string se;
        transform(s.begin(),s.end(),s.begin(),[](unsigned char c){
            return tolower(c);
        });
        for(auto x:s){
            //char ch = tolower(x);
            if((x>='a'&&x<='z')||(x>='0'&&x<='9'))
                se.push_back(x);
            
        }
        int left=0,right=se.length()-1;
        while(left<right){
            if(se[left]!=se[right]) return false;
            left++;
            right--;
        }
        cout<<se<<endl;
        return true;
    }
};

File: 0125-valid-palindrome/0125-valid-palindrome.java
class Solution {
    public boolean isPalli(String s){
        int left=0;
        int right=s.length()-1;
        while(left<=right){
            if(s.charAt(left)!=s.charAt(right)){
                return false;
            }
            left++;
            right--;
        }
        return true;
    }
    public boolean isPalindrome(String s) {
        String ans = "";
        for(int i=0;i<s.length();i++){
            char ch = Character.toLowerCase(s.charAt(i));
            if(Character.isLetter(ch)||Character.isDigit(ch)){
                ans = ans+ch;
            }
        }
        return isPalli(ans);
    }
}